test_sqlitecachedbhandler_torrents has uncaught exceptions

The following uncaught exception occur when running these tests (on `next`):

`test_sqlitecachedbhandler_torrents.py::TestTorrentDBHandler.addTorrent`
```python
Exception in thread Twisted (most likely raised during interpreter shutdown):
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
  File "/usr/lib/python2.7/threading.py", line 754, in run
  File "/home/quinten/Documents/tribler/Tribler/Test/twisted_thread.py", line 53, in <lambda>
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1208, in mainLoop
<type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'msg'
```

`test_sqlitecachedbhandler_torrents.py::TestTorrentDBHandler.updateTorrent` (only fails if the single test is run)
```python
Failure
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/quinten/Documents/tribler/Tribler/dispersy/util.py", line 46, in helper
    return blockingCallFromThread(reactor, func, *args, **kargs)
  File "/home/quinten/Documents/tribler/Tribler/dispersy/util.py", line 240, in blockingCallFromThread
    result.raiseException()
  File "<string>", line 2, in raiseException
AssertionError: None != u'Videoclips'
```

`test_sqlitecachedbhandler_torrents.py::TestTorrentDBHandler.test_add_external_torrent_no_def_invalid`
```python
  File "/home/quinten/Documents/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 342, in addExternalTorrentNoDef
    torrentdef = TorrentDef.load_from_dict(metainfo)
  File "/home/quinten/Documents/tribler/Tribler/Core/TorrentDef.py", line 166, in load_from_dict
    return TorrentDef._create(metainfo)
  File "/home/quinten/Documents/tribler/Tribler/Core/TorrentDef.py", line 119, in _create
    validTorrentFile(metainfo)
  File "/home/quinten/Documents/tribler/Tribler/Core/Utilities/utilities.py", line 138, in validTorrentFile
    raise ValueError('info files length is not int, but ' + repr(type(l)))
ValueError: info files length is not int, but <type 'dict'>
```

Running the entire file will simply report `23/23 tests passed`.

--- a/Tribler/Core/CacheDB/SqliteCacheDBHandler.py
+++ b/Tribler/Core/CacheDB/SqliteCacheDBHandler.py
@@ -350,8 +350,8 @@ class TorrentDBHandler(BasicDBHandler):
                 sql_insert_files = "INSERT OR IGNORE INTO TorrentFiles (torrent_id, path, length) VALUES (?,?,?)"
                 self._db.executemany(sql_insert_files, insert_files)
             except:
-                self._logger.error("Could not create a TorrentDef instance %r %r %r %r %r %r", infohash, timestamp, name, files, trackers, extra_info)
-                print_exc()
+                self._logger.exception("Could not create a TorrentDef instance %r %r %r %r %r %r",
+                                       infohash, timestamp, name, files, trackers, extra_info)
 
     def addOrGetTorrentID(self, infohash):
         assert isinstance(infohash, str), "INFOHASH has invalid type: %s" % type(infohash)

--- a/Tribler/Test/Core/CreditMining/test_creditmining_sys.py
+++ b/Tribler/Test/Core/CreditMining/test_creditmining_sys.py
@@ -6,6 +6,7 @@ Author(s): Ardhi Putra
 import binascii
 import os
 import shutil
+from unittest import skip
 from twisted.internet import defer
 from twisted.internet.defer import inlineCallbacks
 from twisted.internet.task import LoopingCall
@@ -141,6 +142,7 @@ class TestBoostingManagerSys(TestAsServer):
         return defer_param
 
 
+@skip("Credit mining tests are not reliable")
 class TestBoostingManagerSysRSS(TestBoostingManagerSys):
     """
     testing class for RSS (dummy) source
@@ -222,6 +224,7 @@ class TestBoostingManagerSysRSS(TestBoostingManagerSys):
         return self.rss_error_deferred
 
 
+@skip("Credit mining tests are not reliable")
 class TestBoostingManagerSysDir(TestBoostingManagerSys):
     """
     testing class for directory source
@@ -270,6 +273,7 @@ class TestBoostingManagerSysDir(TestBoostingManagerSys):
         return d
 
 
+@skip("Credit mining tests are not reliable")
 class TestBoostingManagerSysChannel(TestBoostingManagerSys, BaseTestChannel):
     """
     testing class for channel source

--- a/Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py
+++ b/Tribler/Test/Core/test_sqlitecachedbhandler_torrents.py
@@ -38,55 +38,6 @@ class TestTorrentFullSessionDBHandler(AbstractDB):
 
 class TestTorrentDBHandler(AbstractDB):
 
-    def setUpPreSession(self):
-        super(TestTorrentDBHandler, self).setUpPreSession()
-        self.config.set_megacache(True)
-        self.config.set_torrent_store(True)
-
-    @blocking_call_on_reactor_thread
-    @inlineCallbacks
-    def setUp(self):
-        yield super(TestTorrentDBHandler, self).setUp()
-
-        from Tribler.Core.APIImplementation.LaunchManyCore import TriblerLaunchMany
-        from Tribler.Core.Modules.tracker_manager import TrackerManager
-        self.session.lm = TriblerLaunchMany()
-        self.session.lm.tracker_manager = TrackerManager(self.session)
-        self.session.lm.tracker_manager.initialize()
-        self.tdb = TorrentDBHandler(self.session)
-        self.tdb.torrent_dir = TESTS_DATA_DIR
-        self.tdb.category = Category()
-        self.tdb.mypref_db = MyPreferenceDBHandler(self.session)
-
-    @blocking_call_on_reactor_thread
-    @inlineCallbacks
-    def tearDown(self):
-        self.tdb.mypref_db.close()
-        self.tdb.mypref_db = None
-        self.tdb.close()
-        self.tdb = None
-
-        yield super(TestTorrentDBHandler, self).tearDown()
-
-    @blocking_call_on_reactor_thread
-    def test_hasTorrent(self):
-        infohash_str = 'AA8cTG7ZuPsyblbRE7CyxsrKUCg='
-        infohash = str2bin(infohash_str)
-        self.assertTrue(self.tdb.hasTorrent(infohash))
-        self.assertTrue(self.tdb.hasTorrent(infohash)) # cache will trigger
-        fake_infohash = 'fake_infohash_100000'
-        self.assertFalse(self.tdb.hasTorrent(fake_infohash))
-
-    @blocking_call_on_reactor_thread
-    def test_get_infohash(self):
-        self.assertTrue(self.tdb.getInfohash(1))
-        self.assertFalse(self.tdb.getInfohash(1234567))
-
-    @blocking_call_on_reactor_thread
-    def test_add_update_Torrent(self):
-        self.addTorrent()
-        self.updateTorrent()
-
     @blocking_call_on_reactor_thread
     def addTorrent(self):
         old_size = self.tdb.size()
@@ -94,9 +45,6 @@ class TestTorrentDBHandler(AbstractDB):
 
         s_infohash = unhexlify('44865489ac16e2f34ea0cd3043cfd970cc24ec09')
         m_infohash = unhexlify('ed81da94d21ad1b305133f2726cdaec5a57fed98')
-
-        sid = self.tdb.getTorrentID(s_infohash)
-        mid = self.tdb.getTorrentID(m_infohash)
 
         single_torrent_file_path = os.path.join(self.getStateDir(), 'single.torrent')
         multiple_torrent_file_path = os.path.join(self.getStateDir(), 'multiple.torrent')
@@ -158,7 +106,6 @@ class TestTorrentDBHandler(AbstractDB):
 
     @blocking_call_on_reactor_thread
     def updateTorrent(self):
-        s_infohash = unhexlify('44865489ac16e2f34ea0cd3043cfd970cc24ec09')
         m_infohash = unhexlify('ed81da94d21ad1b305133f2726cdaec5a57fed98')
         self.tdb.updateTorrent(m_infohash, relevance=3.1415926, category=u'Videoclips',
                                status=u'good', seeder=123, leecher=321,
@@ -175,6 +122,55 @@ class TestTorrentDBHandler(AbstractDB):
         self.assertEqual(leecher, 321)
         last_tracker_check = self.tdb.getOne('last_tracker_check', torrent_id=multiple_torrent_id)
         self.assertEqual(last_tracker_check, 1234567)
+
+    def setUpPreSession(self):
+        super(TestTorrentDBHandler, self).setUpPreSession()
+        self.config.set_megacache(True)
+        self.config.set_torrent_store(True)
+
+    @blocking_call_on_reactor_thread
+    @inlineCallbacks
+    def setUp(self):
+        yield super(TestTorrentDBHandler, self).setUp()
+
+        from Tribler.Core.APIImplementation.LaunchManyCore import TriblerLaunchMany
+        from Tribler.Core.Modules.tracker_manager import TrackerManager
+        self.session.lm = TriblerLaunchMany()
+        self.session.lm.tracker_manager = TrackerManager(self.session)
+        self.session.lm.tracker_manager.initialize()
+        self.tdb = TorrentDBHandler(self.session)
+        self.tdb.torrent_dir = TESTS_DATA_DIR
+        self.tdb.category = Category()
+        self.tdb.mypref_db = MyPreferenceDBHandler(self.session)
+
+    @blocking_call_on_reactor_thread
+    @inlineCallbacks
+    def tearDown(self):
+        self.tdb.mypref_db.close()
+        self.tdb.mypref_db = None
+        self.tdb.close()
+        self.tdb = None
+
+        yield super(TestTorrentDBHandler, self).tearDown()
+
+    @blocking_call_on_reactor_thread
+    def test_hasTorrent(self):
+        infohash_str = 'AA8cTG7ZuPsyblbRE7CyxsrKUCg='
+        infohash = str2bin(infohash_str)
+        self.assertTrue(self.tdb.hasTorrent(infohash))
+        self.assertTrue(self.tdb.hasTorrent(infohash)) # cache will trigger
+        fake_infohash = 'fake_infohash_100000'
+        self.assertFalse(self.tdb.hasTorrent(fake_infohash))
+
+    @blocking_call_on_reactor_thread
+    def test_get_infohash(self):
+        self.assertTrue(self.tdb.getInfohash(1))
+        self.assertFalse(self.tdb.getInfohash(1234567))
+
+    @blocking_call_on_reactor_thread
+    def test_add_update_torrent(self):
+        self.addTorrent()
+        self.updateTorrent()
 
     @blocking_call_on_reactor_thread
     def test_add_external_torrent_no_def_existing(self):

